packages/gatsby-parallel-runner/src/processor-queue/implementations/google-functions/index.js

```diff
@@ -0,0 +1,141 @@
+const { PubSub } = require(`@google-cloud/pubsub`)
+const { Storage } = require(`@google-cloud/storage`)
+const fs = require(`fs-extra`)
+const path = require(`path`)
+const log = require(`loglevel`)
+const { topicFor, bucketFor } = require(`./utils`)
+
+const DEFAULT_MAX_PUB_SUB_SIZE = 1024 * 1024 * 5 // 5 Megabyte
+
+class GoogleFunctions {
+  constructor({
+    processorSettings,
+    maxPubSubSize,
+    noSubscription,
+    googleConfig,
+  }) {
+    this.maxPubSubSize = maxPubSubSize || DEFAULT_MAX_PUB_SUB_SIZE
+    const config =
+      googleConfig ||
+      JSON.parse(fs.readFileSync(process.env.GOOGLE_APPLICATION_CREDENTIALS))
+    this.subName = `gatsby-sub-${new Date().getTime()}`
```

Comment:
This will do the same btw:
```suggestion
    this.subName = `gatsby-sub-${Date.now()}`
```